### PR TITLE
feat: add dataDeliveryVote info to genesis state

### PR DESCRIPTION
**Pull request by audtlr24**
- add `DataDeliveryVote` info to genesis state
- change `GetDeal`, `SetDeal` to be consistent with other functions
- `genesis_test` code will be developed later while refactoring all at once

### CODE REVIEW

**Review by inchori — COMMENTED** [review]
Thanks for your work!
I think you need to add `DataDeliveryVote` in `DefaultGenesis()` function and `Validate()` method which is located in `datadeal/types/genesis.go`.

**Review by inchori — APPROVED** [review]
lgtm

**Comment by audtlr24** [comment]
Thanks, I add `DefaultGenesis()` and `Validate()` functions. And I change the verification of `DeliveredCid`, because I think `DeliveredCid` can be empty when the vote option is `VOTE_OPTION_NO`.

File: x/datadeal/genesis.go [hunk 1]
```diff
@@ -30,25 +35,34 @@ func InitGenesis(ctx sdk.Context, k keeper.Keeper, genState types.GenesisState)
...
+	genesis.DataDeliveryVotes = dataDeliveryVotes
 
 	return &types.GenesisState{
-		Deals:          deals,
-		NextDealNumber: nextDealNum,
+		Deals:             deals,
+		NextDealNumber:    nextDealNum,
+		DataSales:         dataSales,
+		DataDeliveryVotes: dataDeliveryVotes,
 	}
```
Comment (H4NLee) [inline x/datadeal/genesis.go:67]: You can just return `genesis`.

**Review by gyuguen — COMMENTED** [review]
LGTM

**Pull request merged**